inbm/dispatcher-agent/dispatcher/sota/sota.py

```diff
@@ -377,10 +400,12 @@ def execute_from_manifest(self,
             else:
                 # Save the log before reboot
                 self._update_logger.status = FAIL
-                self._update_logger.error = ""
                 self._update_logger.save_log()
-                if self.sota_mode != 'download-only':
-                    self._update_logger.save_granular_log_file()
+                # Always save the granular log in TiberOS.
+                # TODO: Remove Mariner when confirmed that TiberOS is in use
+                if self.sota_mode != 'download-only' or detect_os() == LinuxDistType.TiberOS.name \
```

Comment:
same comment here as above (would be clearer to have another `elif` clause...)

Reply:
Added elif clause